forte/data/extractor/attribute_extractor.py

```diff
@@ -0,0 +1,189 @@
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.converter.feature import Feature
+from forte.data.extractor.base_extractor import BaseExtractor
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "AttributeExtractor"
+]
+
+
+class AttributeExtractor(BaseExtractor):
+    r"""AttributeExtractor extracts feature from the attribute of entry.
+
+    Args:
+        config: An instance of `Dict` or
+            :class:`forte.common.configuration.Config`
+
+            attribute (str): Required. The attribute name of the
+                entry from which features will be extracted. For
+                example, "text" attribute of Token.
+    """
+    def __init__(self, config: Union[Dict, Config]):
+        super().__init__(config)
+
+        if "attribute" not in self.config:
+            raise AttributeError("attribute needs to be specified in "
+                                "the configuration of an AttributeExtractor.")
+
+    @property
+    def state(self) -> Dict:
+        state = super().state
+        state.update({
+            "attribute": self.config.attribute
+        })
+        return state
+
+    @classmethod
+    def from_state(cls, state: Dict) \
+        -> Union["BaseExtractor", "AttributeExtractor"]:
+        obj = BaseExtractor.from_state(state)
+        additional = {
+            "attribute": state["attribute"]
+        }
+        obj.config = Config(obj.config, additional,
+                            allow_new_hparam=True)
+        obj.__class__ = cls
+        return obj
+
+    @staticmethod
+    def get_attribute(entry: Entry, attr: str) -> Any:
+        r"""Functionality: Get the attribute from entry. You can
+        overwrite this function if you have sepcial way to get the
+        attribute from entry.
+
+        Args:
+            entry (Entry): An instance of Entry type, where the
+                attribute will be extracted from.
+            attr (str): The name of the attribute.
+
+        Returns:
+            Any. The attribute extracted from entry.
+        """
+        return getattr(entry, attr)
+
+    @staticmethod
+    def set_attribute(entry: Entry, attr: str, value: Any):
+        r"""Functionality: Set the attribute of an entry to value.
+        You can overwrite this function if you have sepcial way to
+        set the attribute.
+
+        Args:
+            entry (Entry): An instance of Entry type, where the
+                attribute will be set.
+            attr (str): The name of the attribute.
+            value (Any): The value to be set for the attribute.
+        """
+        if attr == "text":
+            raise AttributeError("text attribute of entry cannot "
+                                "be changed.")
+        setattr(entry, attr, value)
+
+    def update_vocab(self, pack: DataPack, instance: Annotation):
+        r"""Functionality: Get all attributes of one instance and
```

Comment:
You can remove this `Functionality:` start.

Reply:
Got it.